### PR TITLE
Updated cluster context with validate function

### PR DESCRIPTION
**Pull request by ganchandrasekaran**

### CODE REVIEW

File: infra/v1alpha1/cluster_context.go [hunk 1]
```diff
@@ -64,3 +75,29 @@ type ClusterContextList struct {
 func init() {
 	SchemeBuilder.Register(&ClusterContext{}, &ClusterContextList{})
 }
+
+func ValidateClusterContextSpec(spec *ClusterContextSpec) error {
```
Comment (kispaljr) [inline infra/v1alpha1/cluster_context.go:79]: you may consider turning this into a method:
`func (spec ClusterContextSpec) ValidateClusterContextSpec() error` , or
`func (spec ClusterContext) ValidateClusterContextSpec() error`
but I'm not pushing, it is good as it is, as well

Comment (ganchandrasekaran) [reply infra/v1alpha1/cluster_context.go:79]: I wanted the same, but then wanted to align with interface-api 
https://github.com/nephio-project/api/blob/main/nf_requirements/v1alpha1/interface_interfaces.go#L68

File: infra/v1alpha1/cluster_context.go [hunk 2]
```diff
@@ -64,3 +75,29 @@ type ClusterContextList struct {
...
+	if spec == nil {
+		return fmt.Errorf("spec invalid: %s", errMissingClusterContextSpec)
+	}
+	if *spec.SiteCode == "" {
```
Comment (kispaljr) [inline infra/v1alpha1/cluster_context.go:83]: check if spec.SiteCode was nil, before dereferencing it with *

**Pull request merged**